org.eclipse.scanning.connector.epics/src/org/eclipse/scanning/connector/epics/EpicsV4ConnectorService.java

```diff
@@ -230,24 +230,26 @@ protected MalcolmMessage sendGetMessage(IMalcolmDevice<?> device, MalcolmMessage
 		PvaClientChannel pvaChannel = null;
 		try {
 			PVStructure pvResult = null;
-			pvaChannel = pvaClient.createChannel(device.getName(),"pva");
-	        pvaChannel.issueConnect();
-	        Status status = pvaChannel.waitConnect(REQUEST_TIMEOUT);
-	        if(!status.isOK()) {
-			String errMEssage = "Failed to connect to device '" + device.getName() + "' (" + status.getType() + ": " + status.getMessage() + ")";
-			throw new Exception(errMEssage);
-	        }
+			pvaChannel = pvaClient.createChannel(device.getName(), "pva");
+			pvaChannel.issueConnect();
+			Status status = pvaChannel.waitConnect(REQUEST_TIMEOUT);
+			if (!status.isOK()) {
+				String errMEssage = "Failed to connect to device '" + device.getName() + "' (" + status.getType() + ": "
+						+ status.getMessage() + ")";
+				throw new Exception(errMEssage);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)